cmd/kubermatic-installer/cmd_mirror_images.go

```diff
@@ -198,6 +200,72 @@ func getAddonsPath(ctx context.Context, logger *logrus.Logger, options *MirrorIm
 	return tempDir, nil
 }
 
+// CollectImageMatrix aggregates images for all cluster versions, cloud providers, and CNI plugins,
+// including both Konnectivity and non-Konnectivity configurations.
+func CollectImageMatrix(
+	logger logrus.FieldLogger,
+	clusterVersions []*version.Version,
+	kubermaticConfig *kubermaticv1.KubermaticConfiguration,
+	allAddons map[string]*addonutil.Addon,
+	versions kubermaticversion.Versions,
+	caBundle resources.CABundle,
+	registryPrefix string,
+) ([]string, error) {
+	var imageList []string
+	for _, clusterVersion := range clusterVersions {
+		for _, cloudSpec := range images.GetCloudSpecs() {
+			for _, cniPlugin := range images.GetCNIPlugins() {
+				versionLogger := logger.WithFields(logrus.Fields{
+					"version":     clusterVersion.Version.String(),
+					"provider":    cloudSpec.ProviderName,
+					"cni-plugin":  string(cniPlugin.Type),
+					"cni-version": cniPlugin.Version,
+				})
+
+				versionLogger.Debug("Collecting images…")
+
+				// Collect images without & with Konnectivity, as Konnectivity / OpenVPN can be switched in clusters
+				// at any time. Remove the non-Konnectivity option once OpenVPN option is finally removed.
+
+				imagesWithoutKonnectivity, err := images.GetImagesForVersion(
+					versionLogger,
+					clusterVersion,
+					cloudSpec,
+					cniPlugin,
+					false,
+					kubermaticConfig,
+					allAddons,
+					versions,
+					caBundle,
+					registryPrefix,
+				)
+				if err != nil {
+					return nil, fmt.Errorf("failed to get images: %w", err)
+				}
+				imageList = append(imageList, imagesWithoutKonnectivity...)
```

Comment:
Just so you know, this is not required anymore since OpenVPN is no longer supported.